### DISCUSSION TITLE
swap api

### DISCUSSION THREAD

**Discussion by jafar-jabr** [reown-com/appkit#2662] [Q&A]
is there an api for calling swap directly from a button without opening the modal all the time ?

**Comment by jafar-jabr** [comment]
@chris13524

**Reply by Rizwankmc** [reply]
you can yse the useAppKit() hook to get the open function and call the open function with -
`const { open } = useAppKit();
  open({view: 'Swap'})
`

**Comment by Rizwankmc** [comment]
its open the swap modal but its making issue while running approve error - 
`{
    "eventId": "417be89e-479c-49cb-b04e-0a701db799e9",
    "url": "http://localhost:3000/profile",
    "domain": "localhost",
    "timestamp": 1742794385305,
    "props": {
        "type": "track",
        "event": "SWAP_APPROVAL_ERROR",
        "properties": {
            "message": "underlying network changed (event=\"changed\", network={\"chainId\":null,\"name\":\"unknown\"}, detectedNetwork={\"name\":\"matic\",\"chainId\":137,\"ensAddress\":null}, code=NETWORK_ERROR, version=providers/5.7.2)",
            "network": "eip155:137",
            "swapFromToken": "USDT",
            "swapToToken": "POL",
            "swapFromAmount": "1.26779600000000000000",
            "swapToAmount": "5.806559",
            "isSmartAccount": false
        }
    }
}`

while the whole app is working and proper network is there when you can see the network in the error object but its give this error
